File: game/src/main/java/me/ryleykimmel/brandywine/game/message/listener/LoginHandshakeMessageListener.java
package me.ryleykimmel.brandywine.game.message.listener;

import me.ryleykimmel.brandywine.game.message.LoginHandshakeMessage;
import me.ryleykimmel.brandywine.game.message.LoginHandshakeResponseMessage;
import me.ryleykimmel.brandywine.network.Session;
import me.ryleykimmel.brandywine.network.message.MessageListener;

/**
 * Listener for the {@link LoginHandshakeMessage}.
 */
public final class LoginHandshakeMessageListener implements MessageListener<LoginHandshakeMessage> {

  /**
   * The status sent upon successful handshake.
   */
  private static final int STATUS_EXCHANGE_DATA = 0;

  /**
   * The client expects a dummy byte array of 8 bytes followed by the session id.
   */
  private static final byte[] EXPECTED_DUMMY = {0, 0, 0, 0, 0, 0, 0, 0};

  @Override
  public void handle(Session session, LoginHandshakeMessage message) {
    session.voidWriteAndFlush(
        new LoginHandshakeResponseMessage(STATUS_EXCHANGE_DATA, EXPECTED_DUMMY, session.getId()));
  }

}

File: game/src/main/kotlin/plugin/message/LoginHandshakeMessageListener.kt
package plugin.message

import me.ryleykimmel.brandywine.game.msg.LoginHandshakeMessage
import me.ryleykimmel.brandywine.game.msg.LoginHandshakeResponseMessage
import me.ryleykimmel.brandywine.network.Session

class LoginHandshakeMessageListener : MessageListener<Session, LoginHandshakeMessage> {

    val STATUS_EXCHANGE_DATA = 0
    val EXPECTED_DUMMY = byteArrayOf(0, 0, 0, 0, 0, 0, 0, 0)

    override fun handle(source: Session, message: LoginHandshakeMessage) {
        source.voidWriteAndFlush(LoginHandshakeResponseMessage(STATUS_EXCHANGE_DATA, EXPECTED_DUMMY, source.sessionKey))
    }

}
